src/block/BaseBigDripleaf.php

```diff
@@ -0,0 +1,136 @@
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\block;
+
+use pocketmine\block\utils\HorizontalFacingTrait;
+use pocketmine\block\utils\SupportType;
+use pocketmine\event\block\StructureGrowEvent;
+use pocketmine\item\Fertilizer;
+use pocketmine\item\Item;
+use pocketmine\math\Facing;
+use pocketmine\math\Vector3;
+use pocketmine\player\Player;
+use pocketmine\world\BlockTransaction;
+
+abstract class BaseBigDripleaf extends Transparent{
+	use HorizontalFacingTrait;
+
+	abstract protected function isHead() : bool;
+
+	private function canBeSupportedBy(Block $block) : bool{
+		//TODO: Moss block
+		return
+			$block instanceof BaseBigDripleaf ||
+			$block->getTypeId() === BlockTypeIds::CLAY ||
+			$block->hasTypeTag(BlockTypeTags::DIRT) ||
+			$block->hasTypeTag(BlockTypeTags::MUD);
+	}
+
+	public function onNearbyBlockChange() : void{
+		if(
+			(!$this->isHead() && !$this->getSide(Facing::UP) instanceof BaseBigDripleaf) ||
+			!$this->canBeSupportedBy($this->getSide(Facing::DOWN))
+		){
+			$this->position->getWorld()->useBreakOn($this->position);
+		}
+	}
+
+	public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
+		$block = $blockReplace->getSide(Facing::DOWN);
+		if(!$this->canBeSupportedBy($block)){
+			return false;
+		}
+		if($player !== null){
+			$this->facing = Facing::opposite($player->getHorizontalFacing());
+		}
+		if($block instanceof BaseBigDripleaf){
+			$this->facing = $block->getFacing();
+			$tx->addBlock($block->getPosition(), VanillaBlocks::BIG_DRIPLEAF_STEM()->setFacing($this->facing));
+		}
+		return parent::place($tx, $item, $blockReplace, $blockClicked, $face, $clickVector, $player);
+	}
+
+	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null, array &$returnedItems = []) : bool{
+		if($item instanceof Fertilizer && $this->grow($player)){
+			$item->pop();
+			return true;
+		}
+		return false;
+	}
+
+	private function seekToHead() : ?BaseBigDripleaf{
+		if($this->isHead()){
+			return $this;
+		}
+		$step = 1;
+		while(($next = $this->getSide(Facing::UP, $step)) instanceof BaseBigDripleaf){
+			if($next->isHead()){
+				return $next;
+			}
+			$step++;
+		}
+		return null;
+	}
+
+	private function grow(?Player $player) : bool{
+		$head = $this->seekToHead();
+		if($head === null){
+			return false;
+		}
+		$pos = $head->getPosition();
+		$up = $pos->up();
+		$world = $pos->getWorld();
+		if(
+			!$world->isInWorld($up->getFloorX(), $up->getFloorY(), $up->getFloorZ()) ||
+			$world->getBlock($up)->getTypeId() !== BlockTypeIds::AIR
+		){
+			return false;
+		}
+
+		$tx = new BlockTransaction($world);
+
+		$tx->addBlock($pos, VanillaBlocks::BIG_DRIPLEAF_STEM()->setFacing($head->getFacing()));
+		$tx->addBlock($up, VanillaBlocks::BIG_DRIPLEAF()->setFacing($head->getFacing()));
```

Comment:
what about the tilt?

Reply:
In vanilla tilt restores to NONE when block grows

Reply:
honestly this sounds like a bug. have you tested it in java?

Reply:
honestly this sounds like a bug. have you tested in Java?

Reply:
Tested it on Java a minute ago, Java seems to be also work this way.

https://www.youtube.com/watch?v=3tPl22WUuXE

Reply:
Fair enough, I guess this can be chalked up to another Mojang inconsistency.